bigcache.go

```diff
@@ -137,6 +137,34 @@ func (c *BigCache) Get(key string) ([]byte, error) {
 	return shard.get(key, hashedKey)
 }
 
+// GetMulti reads entry for each of the keys.
+// It returns an ErrEntryNotFound when
+// no entry exists for the given key.
+func (c *BigCache) GetMulti(keys []string) ([][]byte, error) {
+
+    entries := make([][]byte,len(keys))
+    var err error
+    shardsLocked := make(map[uint64]bool)
+
+    for i := range keys{
+        hashedKey := c.hash.Sum64(keys[i])
+        shard := c.getShard(hashedKey)
+
+        if !shardsLocked[hashedKey & c.shardMask]{
+            shard.lock.RLock()
+            shardsLocked[hashedKey & c.shardMask] = true
+            defer shard.lock.RUnlock()
```

Comment:
Why `defer` ?

Reply:
I tried using the map to RUnlock however i didn't see a difference in performance, it's only called maximum once per shard.

Would doing something like the following be prefered?
```go
for k := range shardsLocked {
    c.shards[k].lock.RUnlock()
}
```